<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="adtteSpec" timestamp="2024-05-13T10:40:05Z" hostname="7955f1639218" tests="22" skipped="1" failures="0" errors="0" time="7.723">
    <testcase time="5.949" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_a_single_gene"/>
    <testcase time="0.016" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_a_single_gene"/>
    <testcase time="0.211" classname="adtteSpec" name="h_km_mae_to_adtte_function_also_works_when_some_ID_variables_are_factors"/>
    <testcase time="0.00199999999999978" classname="adtteSpec" name="h_km_mae_to_adtte_function_also_works_when_some_ID_variables_are_factors"/>
    <testcase time="0.219" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_multiple_genes"/>
    <testcase time="0.00199999999999978" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_multiple_genes"/>
    <testcase time="0.708" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_a_gene_signature"/>
    <testcase time="0.000999999999999446" classname="adtteSpec" name="h_km_mae_to_adtte_function_works_as_expected_with_a_gene_signature"/>
    <testcase time="0.0140000000000011" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_with_invalid_settings"/>
    <testcase time="0.048" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_with_invalid_settings"/>
    <testcase time="0.00499999999999901" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_with_invalid_settings"/>
    <testcase time="0.0440000000000005" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_with_invalid_settings"/>
    <testcase time="0.173" classname="adtteSpec" name="h_km_mae_to_adtte_warns_when_patients_are_not_in_ADTTE_and_therefore_removed"/>
    <testcase time="0.15" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_if_USUBJID_in_MAE_colData_is_different_from_sample_map"/>
    <testcase time="0.126999999999999" classname="adtteSpec" name="h_km_mae_to_adtte_fails_as_expected_if_USUBJID_in_experiment_colData_is_different_from_sample_map"/>
    <testcase time="0.0240000000000009" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.00200000000000067" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="adtteSpec" name="adtteSpecInput_creates_expected_HTML"/>
    <testcase time="0.0239999999999991" classname="adtteSpec" name="adtteSpecServer_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-adtteSpec.R:193:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="assaySpec" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="7" skipped="1" failures="0" errors="0" time="0.042">
    <testcase time="0.0329999999999995" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="assaySpec" name="assaySpecInput_creates_expected_HTML"/>
    <testcase time="0.00399999999999956" classname="assaySpec" name="assaySpecServer_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-assaySpec.R:25:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="barplot" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.128">
    <testcase time="0.122" classname="barplot" name="ui_g_barplot_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="barplot" name="ui_g_barplot_creates_expected_HTML"/>
    <testcase time="0.00499999999999901" classname="barplot" name="barplot_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-barplot.R:26:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="boxplot" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="2" skipped="1" failures="0" errors="0" time="0.044">
    <testcase time="0.0389999999999997" classname="boxplot" name="ui_g_boxplot_creates_expected_HTML"/>
    <testcase time="0.00500000000000078" classname="boxplot" name="boxplot_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-boxplot.R:22:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="checkmate" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="7" skipped="0" failures="0" errors="0" time="0.04">
    <testcase time="0.016" classname="checkmate" name="check_tag_works_as_expected"/>
    <testcase time="0.00200000000000067" classname="checkmate" name="check_tag_works_as_expected"/>
    <testcase time="0.00299999999999834" classname="checkmate" name="check_tag_works_as_expected"/>
    <testcase time="0.00500000000000078" classname="checkmate" name="check_tag_works_as_expected"/>
    <testcase time="0.00500000000000078" classname="checkmate" name="assert_adtte_vars_works_as_expected"/>
    <testcase time="0.00399999999999956" classname="checkmate" name="assert_adtte_vars_works_as_expected"/>
    <testcase time="0.00499999999999901" classname="checkmate" name="assert_adtte_vars_works_as_expected"/>
  </testsuite>
  <testsuite name="experimentSpec" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="10" skipped="1" failures="0" errors="0" time="0.143">
    <testcase time="0.0670000000000002" classname="experimentSpec" name="experimentSpecInput_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="experimentSpec" name="experimentSpecInput_creates_expected_HTML"/>
    <testcase time="0.00699999999999967" classname="experimentSpec" name="h_order_genes_works_as_expected_to_sort_only_by_name"/>
    <testcase time="0.00600000000000023" classname="experimentSpec" name="h_order_genes_works_as_expected_to_sort_only_by_id"/>
    <testcase time="0.00600000000000023" classname="experimentSpec" name="h_order_genes_works_as_expected_with_mixed_id_and_name_sorting"/>
    <testcase time="0.00599999999999845" classname="experimentSpec" name="h_order_genes_does_not_fail_when_given_empty_data_frame"/>
    <testcase time="0.0289999999999999" classname="experimentSpec" name="h_gene_data_works_as_expected"/>
    <testcase time="0.0140000000000011" classname="experimentSpec" name="h_gene_data_does_not_fail_when_object_does_not_contain_any_genes"/>
    <testcase time="0.00300000000000011" classname="experimentSpec" name="h_gene_data_does_not_fail_when_object_does_not_contain_any_genes"/>
    <testcase time="0.00399999999999956" classname="experimentSpec" name="experimentSpec_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-experimentSpec.R:96:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="forest" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.041">
    <testcase time="0.0359999999999996" classname="forest" name="ui_g_forest_tte_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="forest" name="ui_g_forest_tte_creates_expected_HTML"/>
    <testcase time="0.00399999999999956" classname="forest" name="forest_tte_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-forest.R:26:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="geneSpec" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="4" skipped="1" failures="0" errors="0" time="0.037">
    <testcase time="0.0199999999999996" classname="geneSpec" name="geneSpecInput_creates_expected_HTML"/>
    <testcase time="0.00699999999999967" classname="geneSpec" name="h_parse_genes_works_as_expected"/>
    <testcase time="0.00600000000000023" classname="geneSpec" name="h_parse_genes_correctly_returns_empty_data_frame_when_no_genes_match"/>
    <testcase time="0.00399999999999956" classname="geneSpec" name="geneSpec_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-geneSpec.R:39:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="km" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.042">
    <testcase time="0.0360000000000014" classname="km" name="ui_g_km_creates_expected_HTML"/>
    <testcase time="0.00199999999999889" classname="km" name="ui_g_km_creates_expected_HTML"/>
    <testcase time="0.00399999999999956" classname="km" name="km_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-km.R:27:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="pca" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="2" skipped="1" failures="0" errors="0" time="0.084">
    <testcase time="0.0630000000000006" classname="pca" name="ui_g_pca_creates_HTML"/>
    <testcase time="0.0210000000000008" classname="pca" name="pca_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-pca.R:21:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="quality" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.044">
    <testcase time="0.0380000000000003" classname="quality" name="ui_g_quality_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="quality" name="ui_g_quality_creates_expected_HTML"/>
    <testcase time="0.00499999999999901" classname="quality" name="quality_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-quality.R:23:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="sampleVarSpec" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="14" skipped="1" failures="0" errors="2" time="0.193">
    <testcase time="0.0180000000000007" classname="sampleVarSpec" name="sampleVarSpecInput_creates_expected_HTML"/>
    <testcase time="0.000999999999999446" classname="sampleVarSpec" name="sampleVarSpecInput_creates_expected_HTML"/>
    <testcase time="0.00600000000000023" classname="sampleVarSpec" name="h_assign_to_group_list_works_as_expected"/>
    <testcase time="0.00600000000000023" classname="sampleVarSpec" name="h_assign_to_group_list_retains_the_order_of_the_collapsed_levels"/>
    <testcase time="0.00600000000000023" classname="sampleVarSpec" name="h_collapse_levels_works_as_expected_when_a_group_list_is_provided"/>
    <testcase time="0.00799999999999912" classname="sampleVarSpec" name="h_collapse_levels_fails_when_no_group_list_is_provided"/>
    <testcase time="0.00700000000000145" classname="sampleVarSpec" name="h_collapse_levels_fails_when_x_is_not_a_factor_provided"/>
    <testcase time="0.00499999999999901" classname="sampleVarSpec" name="validate_n_levels_passes_for_a_valid_factor_with_required_number_of_levels"/>
    <testcase time="0.00300000000000011" classname="sampleVarSpec" name="validate_n_levels_passes_for_a_valid_factor_with_required_number_of_levels"/>
    <testcase time="0.00699999999999967" classname="sampleVarSpec" name="validate_n_levels_gives_expected_validation_error_when_x_is_not_a_factor"/>
    <testcase time="0.0080000000000009" classname="sampleVarSpec" name="validate_n_levels_gives_expected_validation_error_when_factor_x_does_not_have_required_number_of_levels"/>
    <testcase time="0.0359999999999996" classname="sampleVarSpec" name="sampleVarSpecServer_only_gives_atomic_columns_with_at_least_one_value_in_col_data_vars">
      <error type="error" message="Error in `logger::log_shiny_input_changes(input, namespace = &quot;teal.modules.hermes&quot;)`: No Shiny app running, it makes no sense to call this function outside of a Shiny app ('test-sampleVarSpec.R:129:3')">Error in `logger::log_shiny_input_changes(input, namespace = "teal.modules.hermes")`: No Shiny app running, it makes no sense to call this function outside of a Shiny app
Backtrace:
     x
  1. \-shiny::testServer(...) at test-sampleVarSpec.R:129:3
  2.   +-shiny:::withMockContext(session, rlang::exec(app, !!!args))
  3.   | +-shiny::isolate(...)
  4.   | | +-shiny::..stacktraceoff..(...)
  5.   | | \-ctx$run(...)
  6.   | |   +-promises::with_promise_domain(...)
  7.   | |   | \-domain$wrapSync(expr)
  8.   | |   +-shiny::withReactiveDomain(...)
  9.   | |   | \-promises::with_promise_domain(...)
 10.   | |   |   \-domain$wrapSync(expr)
 11.   | |   |     \-base::force(expr)
 12.   | |   \-env$runWith(self, func)
 13.   | |     \-shiny (local) contextFunc()
 14.   | |       \-shiny::..stacktraceon..(expr)
 15.   | +-shiny::withReactiveDomain(...)
 16.   | | \-promises::with_promise_domain(...)
 17.   | |   \-domain$wrapSync(expr)
 18.   | |     \-base::force(expr)
 19.   | \-withr::with_options(...)
 20.   |   \-base::force(code)
 21.   +-rlang::exec(app, !!!args)
 22.   \-teal.modules.hermes (local) `&lt;fn&gt;`(...)
 23.     \-shiny::moduleServer(...)
 24.       +-session$setReturned(callModule(module, id, session = session))
 25.       \-shiny::callModule(module, id, session = session)
 26.         +-shiny::withReactiveDomain(...)
 27.         | \-promises::with_promise_domain(...)
 28.         |   \-domain$wrapSync(expr)
 29.         |     \-base::force(expr)
 30.         \-teal.modules.hermes (local) module(childScope$input, childScope$output, childScope, ...)
 31.           \-logger::log_shiny_input_changes(input, namespace = "teal.modules.hermes")</error>
    </testcase>
    <testcase time="0.048" classname="sampleVarSpec" name="sampleVarSpecServer_only_gives_factor_columns_in_col_data_vars_when_categorical_only">
      <error type="error" message="Error in `logger::log_shiny_input_changes(input, namespace = &quot;teal.modules.hermes&quot;)`: No Shiny app running, it makes no sense to call this function outside of a Shiny app ('test-sampleVarSpec.R:154:3')">Error in `logger::log_shiny_input_changes(input, namespace = "teal.modules.hermes")`: No Shiny app running, it makes no sense to call this function outside of a Shiny app
Backtrace:
     x
  1. \-shiny::testServer(...) at test-sampleVarSpec.R:154:3
  2.   +-shiny:::withMockContext(session, rlang::exec(app, !!!args))
  3.   | +-shiny::isolate(...)
  4.   | | +-shiny::..stacktraceoff..(...)
  5.   | | \-ctx$run(...)
  6.   | |   +-promises::with_promise_domain(...)
  7.   | |   | \-domain$wrapSync(expr)
  8.   | |   +-shiny::withReactiveDomain(...)
  9.   | |   | \-promises::with_promise_domain(...)
 10.   | |   |   \-domain$wrapSync(expr)
 11.   | |   |     \-base::force(expr)
 12.   | |   \-env$runWith(self, func)
 13.   | |     \-shiny (local) contextFunc()
 14.   | |       \-shiny::..stacktraceon..(expr)
 15.   | +-shiny::withReactiveDomain(...)
 16.   | | \-promises::with_promise_domain(...)
 17.   | |   \-domain$wrapSync(expr)
 18.   | |     \-base::force(expr)
 19.   | \-withr::with_options(...)
 20.   |   \-base::force(code)
 21.   +-rlang::exec(app, !!!args)
 22.   \-teal.modules.hermes (local) `&lt;fn&gt;`(...)
 23.     \-shiny::moduleServer(...)
 24.       +-session$setReturned(callModule(module, id, session = session))
 25.       \-shiny::callModule(module, id, session = session)
 26.         +-shiny::withReactiveDomain(...)
 27.         | \-promises::with_promise_domain(...)
 28.         |   \-domain$wrapSync(expr)
 29.         |     \-base::force(expr)
 30.         \-teal.modules.hermes (local) module(childScope$input, childScope$output, childScope, ...)
 31.           \-logger::log_shiny_input_changes(input, namespace = "teal.modules.hermes")</error>
    </testcase>
    <testcase time="0.0340000000000007" classname="sampleVarSpec" name="sampleVarSpec_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-sampleVarSpec.R:173:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="scatterplot" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.044">
    <testcase time="0.0379999999999985" classname="scatterplot" name="ui_g_scatterplot_creates_expected_HTML"/>
    <testcase time="0.00200000000000067" classname="scatterplot" name="ui_g_scatterplot_creates_expected_HTML"/>
    <testcase time="0.00399999999999956" classname="scatterplot" name="scatterplot_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-scatterplot.R:26:3')"/>
    </testcase>
  </testsuite>
  <testsuite name="utils" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="6" skipped="0" failures="0" errors="0" time="0.035">
    <testcase time="0.0150000000000006" classname="utils" name="is_blank_works_as_expected"/>
    <testcase time="0.00300000000000011" classname="utils" name="is_blank_works_as_expected"/>
    <testcase time="0.00200000000000067" classname="utils" name="is_blank_works_as_expected"/>
    <testcase time="0.00799999999999912" classname="utils" name="h_extract_words_works_as_expected"/>
    <testcase time="0.00200000000000067" classname="utils" name="h_extract_words_works_as_expected"/>
    <testcase time="0.00499999999999901" classname="utils" name="h_extract_words_works_as_expected"/>
  </testsuite>
  <testsuite name="volcanoplot" timestamp="2024-05-13T10:40:13Z" hostname="7955f1639218" tests="3" skipped="1" failures="0" errors="0" time="0.056">
    <testcase time="0.0510000000000002" classname="volcanoplot" name="ui_g_volcanoplot_creates_expected_HTML"/>
    <testcase time="0.00100000000000122" classname="volcanoplot" name="ui_g_volcanoplot_creates_expected_HTML"/>
    <testcase time="0.00399999999999956" classname="volcanoplot" name="volcanoplot_module_works_as_expected_in_the_test_app">
      <skipped message="Reason: testing depth 3 is below current testing specification 5 ('test-volcanoplot.R:21:3')"/>
    </testcase>
  </testsuite>
</testsuites>
